lib/query/counts/genEndpoint.server.js

```diff
@@ -0,0 +1,70 @@
+import { check } from 'meteor/check';
+import { Meteor } from 'meteor/meteor';
+import { Mongo } from 'meteor/mongo';
+
+import { COUNTS_COLLECTION_CLIENT } from './constants';
+
+// XXX: Should this persist between server restarts?
+const collection = new Mongo.Collection(null);
+
+/**
+ * This method generates a reactive count endpoint (a method and publication) for a collection or named query.
+ *
+ * @param {String} name - Name of the query or collection
+ * @param {Function} getCursor - Takes in the user's session document as an argument, and turns that into a Mongo cursor.
+ * @param {Function} getSession - Takes the subscribe method's argument as its parameter. Should enforce any necessary security constraints. The return value of this function is stored in the session document.
+ */
+export default (name, { getCursor, getSession }) => {
+    Meteor.methods({
+        [name + '.count.subscribe'](paramsOrBody) {
+            const session = getSession.call(this, paramsOrBody);
+            const existingSession = collection.findOne({ ...session, userId: this.userId });
```

Comment:
We should also insert "name" in here, because we may have collisions in future, if we have 2 diff collections with same params or filters.